tests/test_utils.py

```diff
@@ -757,6 +757,36 @@ def test_send_project_access_reset_email(client):
     assert response is None
 
 
+# send email functions
+
+
+def test_send_email_with_retry(client):
+    """Send email with retry ok first try."""
+    # Get email row
+    email_row: models.Email = db.session.query(models.Email).first()
+
+    # Run function
+    with patch("dds_web.utils.mail.send"):
+        response = utils.send_email_with_retry(msg=email_row)
+    assert response is None
+
+
+def test_send_email_with_retry_exception(client):
+    """Send email with retry - exception on the 1st try."""
+
+    import smtplib
+
+    side_effect = [smtplib.SMTPException, None]  # Exception on the first try, second try is ok
+
+    # Get email row
+    email_row: models.Email = db.session.query(models.Email).first()
+
+    # Run function
+    with patch("dds_web.utils.mail.send", side_effect=side_effect):
+        response = utils.send_email_with_retry(msg=email_row)
+    assert response is None
+
+
```

Comment:
Answer to your previous question: Yes, remove this too. BUT put it in another branch, it's great that you've added these tests, we definitely want to keep them, just not in this PR.